util/reggen/ip_block.py

```diff
@@ -163,6 +167,11 @@ def from_raw(param_defaults: List[Tuple[str, str]],
                                      .format(name),
                                      rd.get('alert_list', []))
 
+        countermeasures = CounterMeasure.from_raw_list(
+            'countermeasure list for block {}'
+            .format(name),
```

Comment:
Ohh, I now understand what `what` argument is - it's just used as a debug aid in case something went wrong. IMO in that case, perhaps just wire up the block `name` and make that a `Countermeasure` class attribute?

Reply:
Yes exactly. I followed this pattern since it is used everywhere in other mini classes of used in IP block (see for instance `signal.py`, `alert.py`). I suggest to leave this as-is for uniformity.

Reply:
I see, ok. The `from_raw_list` already knows it is attempting to parse a "countermeasure list for block foo", so the only useful information here is "foo". Anyway, fine with cleaning this up later.

Reply:
As a note: this sort of thing is a bit finicky to get right. My usual approach is to try writing nonsense data that triggers each of the checks and make sure that the error messages look sensible.

Needing this is an unfortunate side-effect of using something like the `hjson` (or indeed `yaml`) libraries, where you don't have line information to be able to say "there's something wrong on line 123" and instead you have to say "if you look at your configuration file, and follow keys A, B then C, then the 4th item is malformed". What a pain!

Reply:
There is only 1 hjson file per IP. The IP block name, the input `raw`, and the fact that this code is processing a countermeasure data structure should be sufficient to help the user pinpoint where exactly they made a mistake in the hjson.